server/src/main/java/io/spine/examples/chatspn/server/chat/ChatCardRepository.java

```diff
@@ -0,0 +1,138 @@
+import io.spine.examples.chatspn.chat.event.GroupChatCreated;
+import io.spine.examples.chatspn.chat.event.MembersAdded;
+import io.spine.examples.chatspn.chat.event.MembersRemoved;
+import io.spine.examples.chatspn.chat.event.PersonalChatCreated;
+import io.spine.examples.chatspn.chat.event.UserLeftChat;
+import io.spine.examples.chatspn.message.event.MessageContentUpdated;
+import io.spine.examples.chatspn.message.event.MessageMarkedAsDeleted;
+import io.spine.examples.chatspn.message.event.MessagePosted;
+import io.spine.examples.chatspn.server.ProjectionReader;
+import io.spine.server.projection.ProjectionRepository;
+import io.spine.server.route.EventRouting;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static com.google.common.collect.ImmutableSet.toImmutableSet;
+import static io.spine.client.Filters.eq;
+
+/**
+ * The repository for managing {@link ChatCardProjection} instances.
+ */
+public final class ChatCardRepository
+        extends ProjectionRepository<ChatCardId, ChatCardProjection, ChatCard> {
+
+    @OverridingMethodsMustInvokeSuper
+    @Override
+    protected void setupEventRouting(EventRouting<ChatCardId> routing) {
+        super.setupEventRouting(routing);
+        routing.route(PersonalChatCreated.class,
+                      (event, context) -> toUsersInChat(
+                              ImmutableList.of(event.getCreator(), event.getMember()),
+                              event.getId()))
+               .route(GroupChatCreated.class,
+                      (event, context) -> {
+                          var users = new ArrayList<>(event.getMemberList());
+                          users.add(event.getCreator());
+                          return toUsersInChat(users, event.getId());
+                      })
+               .route(ChatMarkedAsDeleted.class,
+                      (event, context) -> toEverybodyInChat(event.getId(), context))
+               .route(MessagePosted.class,
+                      (event, context) -> toEverybodyInChat(event.getChat(), context))
+               .route(MessageContentUpdated.class,
+                      (event, context) -> toEverybodyInChat(event.getChat(), context))
+               .route(MessageMarkedAsDeleted.class,
+                      (event, context) -> toEverybodyInChat(event.getChat(), context))
+               .route(MembersAdded.class,
+                      (event, context) -> toUsersInChat(event.getMemberList(), event.getId()))
+               .route(MembersRemoved.class,
+                      (event, context) -> toUsersInChat(event.getRemovedMemberList(),
+                                                        event.getId()))
+               .route(UserLeftChat.class,
+                      (event, context) -> toUsersInChat(ImmutableList.of(event.getUser()),
+                                                        event.getChat()));
+    }
+
+    /**
+     * Returns ID's of chat cards of provided users in the chat.
```

Comment:
Please have a read on using apostrophe in English. Right now it's not being used correctly.